core/range-controller.js

```diff
@@ -773,7 +773,13 @@ var RangeController = exports.RangeController = Montage.specialize( /** @lends R
                 // remove all values from the selection that were removed (but
                 // not added back)
                 minus.deleteEach(plus, equals);
-                if (this.selection) {
+
+                if (this.selection.length) {
+                    // ensure selection always has content
+                    if (this.content.length && this.avoidsEmptySelection && !this.multiSelect) {
```

Comment:
This is nested inside another length check:
https://github.com/rayshan/montage/blob/pr-rangecontroller/core/range-controller.js#L771
...which should probably check for `this.selection` existence first, but I didn't want to touch code that wasn't considered broken. There are quite a few other `length` checks in this file that can be rewritten.